plugins/dfu/dfu-tool.c

```diff
@@ -938,8 +938,6 @@ dfu_tool_write (DfuToolPrivate *priv, gchar **values, GError **error)
 	/* do host reset */
 	if (!fu_device_attach (FU_DEVICE (device), error))
 		return FALSE;
-	if (!dfu_device_wait_for_replug (priv, device, FU_DEVICE_REMOVE_DELAY_RE_ENUMERATE, error))
```

Comment:
after device attach it will boot into application mode, so there is no point waiting for device replug, as it won't enter dfu again

Reply:
Right, so isn't this the time it's waiting to enter into application mode as requested a few lines back?

Reply:
well after flashing i get this message and exit code 1
```0483:5740 is not supported: no DFU interfaces```
which makes sense because the device didn't boot into dfu mode.

Reply:
We need these lines for classic DFU devices, e.g. not the STM variant. Would it be possible to be sent on of these devices so we can add it to the hardware text suite to prevent regressions?

Reply:
i added a new quirk for this, attach-into-application